rskj-core/src/main/java/co/rsk/rpc/Web3InformationRetriever.java

```diff
@@ -70,14 +73,22 @@ public Web3InformationRetriever(TransactionPool transactionPool,
      */
     public Optional<Block> getBlock(String identifier) {
         Block block;
-        if (PENDING.equals(identifier)) {
-            block = executionBlockRetriever.retrieveExecutionBlock(identifier).getBlock();
-        } else if (LATEST.equals(identifier)) {
-            block = blockchain.getBestBlock();
-        } else if (EARLIEST.equals(identifier)) {
-            block = blockchain.getBlockByNumber(0);
-        } else {
-            block = this.blockchain.getBlockByNumber(getBlockNumber(identifier));
+
+        switch (identifier) {
+            case PENDING:
+                block = executionBlockRetriever.retrieveExecutionBlock(identifier).getBlock();
+                break;
+            case LATEST:
+                block = blockchain.getBestBlock();
+                break;
+            case EARLIEST:
+                block = blockchain.getBlockByNumber(0);
+                break;
+            default:
+                byte[] hash = getBlockHash(identifier);
```

Comment:
I would at some unit test to check this logic related with blockNumber and blockHash.